lib/ProductOpener/DataQualityFood.pm

```diff
@@ -1373,6 +1373,69 @@ sub check_categories ($product_ref) {
 	return;
 }
 
+=head2 check_labels( PRODUCT_REF )
+
+Checks related to specific product labels.
+
+Vegan label: check that there is no non-vegan ingredient.
+
+Vegetarian label: check that there is no non-vegetarian ingredient.
+
+=cut
+
+sub check_labels ($product_ref) {
+	if (defined $product_ref->{labels_tags} && has_tag($product_ref, "labels", "en:vegan")) {
```

Comment:
Here you check for the vegan label, but then below at line 1420 you have data quality warnings for the vegetarian label